retrofit2-clients/src/main/java/com/palantir/remoting2/retrofit2/MultiServerRetryInterceptor.java

```diff
@@ -86,12 +86,12 @@ public Response intercept(Chain chain) throws IOException {
     }
 
     private Request redirectRequest(Request request, String redirectToUri) {
-        String requestUri = request.url().toString();
+        HttpUrl requestUrl = request.url();
 
         String matchingUri = null;
         // Find which server from `uris` is used in the current request, then ...
         for (String uri : uris) {
```

Comment:
We seem to use uri everywhere including when calling the constructors and the Request object has `url` anyway. I could pass a String to doServersMatch for the prefixUrl, but not sure if that's any cleaner